File: app/src/main/java/com/github/premnirmal/ticker/portfolio/SuggestionsAdapter.java
package com.github.premnirmal.ticker.portfolio;

import android.content.Context;
import android.text.Html;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.github.premnirmal.tickerwidget.R;
import com.github.premnirmal.ticker.network.Suggestion;

import java.util.List;

/**
 * Created by premnirmal on 12/21/14.
 */
class SuggestionsAdapter extends BaseAdapter {

    final List<Suggestion> suggestions;

    SuggestionsAdapter(List<Suggestion> suggestions) {
        this.suggestions = suggestions;
    }

    @Override
    public int getCount() {
        return suggestions.size();
    }

    @Override
    public Suggestion getItem(int position) {
        return suggestions.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        final Context context = parent.getContext();
        if (convertView == null) {
            convertView = new TextView(context);
        }
        final TextView textView = (TextView) convertView;
        final Suggestion item = getItem(position);
        final String name = item.symbol + " - " + Html.fromHtml(item.name) + " (" + item.exchDisp + ")";
        textView.setText(name);
        final int padding = (int) context.getResources().getDimension(R.dimen.text_padding);
        textView.setPadding(padding, padding, padding, padding);

        return textView;
    }
}

File: app/src/main/kotlin/com/github/premnirmal/ticker/portfolio/SuggestionsAdapter.kt
package com.github.premnirmal.ticker.portfolio

import android.text.Html
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import com.github.premnirmal.ticker.network.Suggestion
import com.github.premnirmal.tickerwidget.R

/**
 * Created by premnirmal on 2/26/16.
 */
internal class SuggestionsAdapter(val suggestions: List<Suggestion>) : BaseAdapter() {

    override fun getCount(): Int {
        return suggestions.size
    }

    override fun getItem(position: Int): Suggestion {
        return suggestions[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val context = parent.context
        val textView: TextView
        if (convertView == null) {
            textView = TextView(context)
        } else {
            textView = convertView as TextView
        }
        val item = getItem(position)
        val name = item.symbol + " - " + Html.fromHtml(item.name) + " (" + item.exchDisp + ")"
        textView.text = name
        val padding = context.resources.getDimension(R.dimen.text_padding).toInt()
        textView.setPadding(padding, padding, padding, padding)

        return textView
    }
}
